src/test_helpers/components.rs

```diff
@@ -224,6 +234,42 @@ impl<'a> TestWorkloadBuilder<'a> {
             .namespaces
             .child(&self.w.workload.node, &self.w.workload.name)?;
         self.w.workload.workload_ip = network_namespace.ip();
+
+        for (vip, ports) in &self.w.vips {
```

Comment:
nit: Now that we have services we may want to replace the vip() method with one that adds a full Service instead of reverse engineering it here?